packages/grafana-data/src/utils/location.test.ts

```diff
@@ -1,18 +1,65 @@
 import { locationUtil } from './location';
 
 describe('locationUtil', () => {
+  const { location } = window;
+
   beforeAll(() => {
-    locationUtil.initialize({
-      config: { appSubUrl: '/subUrl' } as any,
-      getVariablesUrlParams: (() => {}) as any,
-      getTimeRangeForUrl: (() => {}) as any,
-    });
+    // @ts-ignore
+    delete window.location;
+
+    window.location = {
+      ...location,
+      hash: '#hash',
+      host: 'www.domain.com:9877',
+      hostname: 'www.domain.com',
+      href: 'http://www.domain.com:9877/path/b?search=a&b=c&d#hash',
+      origin: 'http://www.domain.com:9877',
+      pathname: '/path/b',
+      port: '9877',
+      protocol: 'http:',
+      search: '?search=a&b=c&d',
+    };
+  });
+
+  afterAll(() => {
+    window.location = location;
   });
 
-  describe('With /subUrl as appSubUrl', () => {
-    it('/subUrl should be stripped', () => {
+  describe('strip base when appSubUrl configured', () => {
+    beforeEach(() => {
+      locationUtil.initialize({
+        config: { appSubUrl: '/subUrl' } as any,
+        getVariablesUrlParams: (() => {}) as any,
+        getTimeRangeForUrl: (() => {}) as any,
+      });
+    });
+    test('relative url', () => {
       const urlWithoutMaster = locationUtil.stripBaseFromUrl('/subUrl/grafana/');
       expect(urlWithoutMaster).toBe('/grafana/');
     });
+    test('absolute url url', () => {
+      const urlWithoutMaster = locationUtil.stripBaseFromUrl('http://www.domain.com:9877/subUrl/grafana/');
+      expect(urlWithoutMaster).toBe('/grafana/');
+    });
+  });
+
+  describe('strip base when appSubUrl not configured', () => {
+    beforeEach(() => {
+      locationUtil.initialize({
+        config: {} as any,
+        getVariablesUrlParams: (() => {}) as any,
+        getTimeRangeForUrl: (() => {}) as any,
+      });
+    });
+
+    test('relative url', () => {
+      const urlWithoutMaster = locationUtil.stripBaseFromUrl('/subUrl/grafana/');
+      expect(urlWithoutMaster).toBe('/subUrl/grafana/');
+    });
+
+    test('absolute url', () => {
```

Comment:
This scenario seem incorrect to me, an absolute url should stay absolute

Reply:
Well, yes and no. This function is used for getting everything after the appSubUrl segment in the URL. I think it shouldn't care whether or not the input URL is absolute or relative. Given `[url|/url]/appSubPath/pathname` should just give pathname. Or the input if there is nothing to strip.